Sources/QuranAudioKit/AudioPlayer/GappedAudioRequestBuilder.swift

```diff
@@ -60,9 +62,9 @@ final class GappedAudioRequestBuilder: QuranAudioRequestBuilder {
             let verses = surasDictionary[sura] ?? []
 
             // add besm Allah for all except Al-Fatihah and At-Tawbah
-            if sura.startsWithBesmAllah && verses[0] == sura.firstVerse {
+            if (requestRuns == .one || ayahs.isEmpty == false) && sura.startsWithBesmAllah && verses[0] == sura.firstVerse {
                 urls.append(createRequestInfo(reciter: reciter, verse: start.quran.firstVerse))
-                ayahs.append(verses[0])
+                ayahs.append(start.quran.firstVerse)
```

Comment:
This is wrong, verses[0] is not the same as quran.firstVerse. verses[0] is the start verse of the sura. So for surat Al baqara that's (sura=2, verse=1) while quran.firstVerse is (sura=1, verse=1)